homeassistant/components/habitica/binary_sensor.py

```diff
@@ -0,0 +1,85 @@
+"""Binary sensor platform for Habitica integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from enum import StrEnum
+from typing import Any
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import ASSETS_URL
+from .entity import HabiticaBase
+from .types import HabiticaConfigEntry
+
+
+@dataclass(kw_only=True, frozen=True)
+class HabiticaBinarySensorEntityDescription(BinarySensorEntityDescription):
+    """Habitica Binary Sensor Description."""
+
+    value_fn: Callable[[dict[str, Any]], bool | None]
+    entity_picture: Callable[[dict[str, Any]], str | None]
+
+
+class HabiticaBinarySensor(StrEnum):
+    """Habitica Entities."""
+
+    PENDING_QUEST = "pending_quest"
+
+
+BINARY_SENSOR_DESCRIPTIONS: tuple[HabiticaBinarySensorEntityDescription, ...] = (
+    HabiticaBinarySensorEntityDescription(
+        key=HabiticaBinarySensor.PENDING_QUEST,
+        translation_key=HabiticaBinarySensor.PENDING_QUEST,
+        value_fn=lambda user: user["party"]["quest"]["RSVPNeeded"],
+        entity_picture=(
+            lambda user: f"inventory_quest_scroll_{key}.png"
```

Comment:
This also should be a `def` instead of `lambda`.

Reply:
same as above